dataloader.py

```diff
@@ -30,7 +30,14 @@ def get_vocab(self):
     def get_seq_length(self):
         return self.seq_length
 
+    def read_files(self):
+        self.feats_fc = h5py.File(os.path.join(
+            self.opt.input_fc_dir, 'feats_fc.h5'), 'r')
+        self.feats_att = h5py.File(os.path.join(
+            self.opt.input_att_dir, 'feats_att.h5'), 'r')
+
     def get_npy_data(self, ix):
+        self.read_files()
```

Comment:
why this fix the parallel problem?

Also, if self.feats_{fc,att} aren't needed anymore this function call, I think we should delete them.

Reply:
The problem was that before, the file was only opened once and then all the threads wanted to read from that. Now each time we need access we load the files and there is no thread blocking.

`self.feats_fc` is used but we are not using `self.att`.